packages/core/src/governance/client/hooks/useChangeVotingMethodOrchestrator.ts

```diff
@@ -100,9 +100,18 @@ export const useCreateChangeVotingMethodOrchestrator = ({
   authToken?: string | null;
   config?: Config;
 }) => {
-  const web3 = useChangeVotingMethodMutationsWeb3Rpc(config);
-  const agreementFiles = useAgreementFileUploads(authToken);
   const web2 = useAgreementMutationsWeb2Rsc(authToken);
+  const web3 = useChangeVotingMethodMutationsWeb3Rpc(config);
+  const agreementFiles = useAgreementFileUploads(
+    authToken,
+    (uploadedFiles, slug) => {
+      web2.updateAgreementBySlug({
+        slug: slug ?? '',
+        attachments: uploadedFiles?.attachments,
+        leadImage: uploadedFiles?.leadImage,
+      });
+    },
```

Comment:
Let's create a handler like described in my other comment. Get the `slug` from `web2.createAgreement.slug`. Use `invariant` to throw when `slug` is not defined. It has to be as we're calling the file upload only when the the web2 creation was successful